Modules/_abc.c

```diff
@@ -20,13 +20,25 @@ _Py_IDENTIFIER(_abc_impl);
 _Py_IDENTIFIER(__subclasscheck__);
 _Py_IDENTIFIER(__subclasshook__);
 
+typedef struct {
+    PyObject *_abc_data_type;
+} _abcmodulestate;
+
 /* A global counter that is incremented each time a class is
    registered as a virtual subclass of anything.  It forces the
    negative cache to be cleared before its next use.
    Note: this counter is private. Use `abc.get_cache_token()` for
    external code. */
 static unsigned long long abc_invalidation_counter = 0;
```

Comment:
I am not sure that the abc_invalidation_counter is okay to be shared as static from interpreter Isolation point of view.

Reply:
Should we move this into `PyInterpreterState`?

Reply:
It seems like it would be safe to put it in the module state. The purpose of the counter is to invalidate the module cache. Registering a class in module instance 1 should not invalidate module instance 2. What do you think?

Reply:
I agree, I 've updated the PR.